common/attachments/index.ts

```diff
@@ -0,0 +1,52 @@
+import {Student} from "../entity/Student";
+import {Pupil} from "../entity/Pupil";
+import { prisma } from '../prisma';
+import { v4 as uuid } from "uuid";
+import {putFile, ATTACHMENT_BUCKET, generatePresignedURL} from "../file-bucket";
+
+export interface AttachmentGroup {
+    attachmentGroupId: string;
+    attachmentListHTML: string;
+    attachmentIds: string[];
+}
+
+/*
+Creates an attachment in the database and uploads the specified file to the S3 bucket.
+
+attachmentGroupId: Unique per group of attachments (per message)
+returns: attachmentId: Unique per individual attachment
+ */
+export async function createAttachment(file: Express.Multer.File, uploader: Student | Pupil, attachmentGroupId: string) {
+    let attachmentId = uuid().toString();
+    await prisma.attachment.create({
+        data: {
+            id: attachmentId,
+            uploaderID: uploader.wix_id,
```

Comment:
Ah, one last thing: Do we want to continue to use the `wix_id`? Do we use that as a user id in other places too?

Reply:
Oh, I was under the impression that we used wix_id, but that's probably because I worked on some older code a few months ago. Changed it to `getUserId(uploader)` 👍

Reply:
(which is what the notification system also uses)

Reply:
Hm, the more I think about this, `getUserId` is not quite optimal as we can't JOIN it on database level (well, we can with some `SUBSTRING(...)` magic. The good thing with the wix_id (apart from it's name) is that it is actually unique across all users, the bad thing is that for retrieving the related user record, one has to query all user tables (student, pupil, screener, mentor). Maybe we should rethink this before shipping this to production (the notification system is currently not used anywhere yet)